packages/appkit/src/client.ts

```diff
@@ -1403,44 +1403,46 @@ export class AppKit {
   }: Pick<AdapterBlueprint.ConnectResult, 'address' | 'chainId'> & {
     chainNamespace: ChainNamespace
   }) {
-    this.setPreferredAccountType(
-      AccountController.state.preferredAccountType
-        ? AccountController.state.preferredAccountType
-        : 'eoa',
-      ChainController.state.activeChain as ChainNamespace
-    )
-
-    this.setCaipAddress(
-      `${chainNamespace}:${chainId}:${address}` as `${ChainNamespace}:${string}:${string}`,
-      chainNamespace
-    )
+    // Only update state when needed
+    if (address !== AccountController.state.address) {
```

Comment:
Is there any chance to happen a miss match due to uppercase and lowercase characters?

Reply:
good catch!

Reply:
there shouldn't change cause we don't change it anywhere, but added a lowercase jic